src/cli.rs

```
@@ -25,6 +27,36 @@ pub struct Arguments {
     pub stderr_file: Option<path::PathBuf>,
 }
 
+pub fn parse_arguments_and_read_file(args: &Arguments) -> Result<(String, Option<String>)> {
+    //TODO: maybe to use sanitization to prevent reading files outside the project directory
+    let source_code = fs::read_to_string(&args.source_file)
+        .with_context(|| format!("could not read sourcefile `{}`", args.source_file.display()))?;
+
+    let err_file =
+        match &args.stderr_file {
+            Some(stderr_file) => Some(fs::read_to_string(stderr_file).with_context(|| {
+                format!("could not read stderr file `{}`", stderr_file.display())
+            })?),
+            None => None,
+        };
+
+    Ok((source_code, err_file))
+}
+
+pub fn update_source_code(args: &Arguments, new_code: String) -> Result<()> {
+    fs::remove_file(&args.source_file)
+        .with_context(|| format!("could not remove file `{}`", &args.source_file.display()))?;
+
+    fs::write(&args.source_file, new_code).with_context(|| {
+        format!(
+            "could not write updated code to file `{}`",
+            &args.source_file.display()
+        )
+    })?;
```

I don't think we want to remove the file and then create a new file at the same location. I think it's a bit confusing if you're using a tool, and it silently replaces your file :) plus, if you are working on the tool and trying it out on a Rust file, it's annoying that this file gets replaced everytime you run the tool. what if you make a mistake and you end up leaving a blank file?

I think a better solution would be to simply output the result on `stdout`, with `println!()` for example. so literally `println!("{new_code}")`

this way, you can run the tool like so:

```
./rusttest-to-dg -f file.rs -e file.stderr > new_file.rs
```